sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java

```diff
@@ -413,8 +413,15 @@ public static String toTableSpec(TableReference ref) {
   }
 
   public static String dataCatalogName(TableReference ref) {
-    return String.format(
-        "bigquery:%s.%s.%s", ref.getProjectId(), ref.getDatasetId(), ref.getTableId());
+    String tableIdBase;
+    int ix = ref.getTableId().indexOf('$');
+    if (ix == -1) {
+      tableIdBase = ref.getTableId();
+    } else {
+      tableIdBase = ref.getTableId().substring(0, ix);
+    }
+    System.out.println("XXX " + ref.getTableId() + " " + ix + tableIdBase);
```

Comment:
is this `System.out.println` a leftover, or one should use slf4j logger?

Reply:
Removed.